Jint/Runtime/Environments/DeclarativeEnvironmentRecord.cs

```diff
@@ -122,6 +122,18 @@ public sealed override JsValue GetBindingValue(string name, bool strict)
             return null;
         }
 
+        internal override bool TryGetBindingValue(string name, bool strict, out JsValue value)
```

Comment:
Keeping the implementations entirely separate from `GetBindingValue`, simply because code reuse would either require catching exceptions from `GetBindingValue` (somewhat defeating the point of `Try...` in the first place). Or `GetBindingValue` would have to make a call to `TryGetBindingValue`, which is what I'd do in just about any other project. Not sure if a call could be hinted as inlined, but if any method in Jint doesn't need additional overhead, it's probably `GetBindingValue` - is there anything in Jint that's called more often during interpretation?

Reply:
Yes this is quite performance critical and inlining could get hard due to virtuals. I think this is the right approach.